File: Data Structures & Algorithms/sqrtx/submission-0.java
class Solution {
    public int mySqrt(int x) {
        int left =0;
        int right = x;
        int res=0;
        while(left <= right){
            int m = left+ (right-left)/2;
            if ((long) m * m > x) {
                right = m - 1;
            } else if ((long) m * m < x) {
                left = m + 1;
                res = m;
            } else {
                return m;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/baseball-game/submission-0.js
class Solution {
    /**
     * @param {string[]} operations
     * @return {number}
     */
    calPoints(operations) {
        let stack=[]
        let res = 0;
        for(let i=0;i<operations.length;i++){
            if(operations[i] === 'C'){
                let val = stack.pop()
                res= res-val
            }else if(operations[i] === 'D'){
                let val =  stack[stack.length-1] * 2
                stack.push(val);
                res += val
            }else if(operations[i] === '+'){
                let val = stack[stack.length-1] + stack[stack.length-2];
                stack.push(val)
                res += val
            }else{
                stack.push(parseInt(operations[i]))
                res += stack[stack.length - 1];
            }
        }
        return res
    }
}
